app/src/main/java/com/github/sdp_begreen/begreen/fragments/SendPostFragment.kt

```diff
@@ -189,11 +199,69 @@ class SendPostFragment : Fragment() {
 
                 }
 
-                // Update the user and return
-                lifecycleScope.launch {
-                    updateUser(metadata, user)
-                    returnToCamera()
-                }
+                // Obtain reference to Firebase's '.info/connected' node to monitor connection state.
+                val connectedRef = Firebase.database.getReference(".info/connected")
+
+                // Attach a listener for a single event to check if client is currently connected to Firebase.
+                connectedRef.addListenerForSingleValueEvent(object : ValueEventListener {
+                    override fun onDataChange(snapshot: DataSnapshot) {
+                        // Retrieve connectivity state from snapshot. If data is null, assume not connected.
+                        val connected = snapshot.getValue(Boolean::class.java) ?: false
+
+                        // If connected, proceed with user update and photo sharing operations.
+                        if (connected) {
+                            Log.d("TAG", "connected")
+                            // Launch coroutine to perform potentially long-running operations off the main UI thread.
+                            lifecycleScope.launch {
+                                // Update the user with the new photo and return to the camera.
+                                updateUser(metadata, user)
+                                returnToCamera()
+                            }
+                        // If not connected, store metadata and user information for later use when connectivity is re-established.
+                        } else {
+                            Log.d("TAG", "not connected")
+
+                            // Use TinyDB for local data storage.
+                            val tinyDB = TinyDB(this@SendPostFragment.context)
+
+                            // Retrieve existing metadata and add current one.
+                            val metas = tinyDB.getListObject("metas", TrashPhotoMetadata::class.java)
```

Comment:
Here, I am not a big fan of having `metas`, `users` and `bitmaps` as hard coded keys, that you have to again hard code in `BeGreenApp` to be able to retrieve the stored values. It's really error prone, and not at all Maintainable.

Instead, I would suggest you to have an `enum` with the possible keys, so that you only have to pass an instance enum to get the correct value, and the key is only defined once in the enum instance.

Let me know what you think of it.

Reply:
Changes made as proposed, thanks!